dartagnan/src/main/java/com/dat3m/dartagnan/utils/options/BaseOptions.java

```diff
@@ -109,26 +110,32 @@ public void parse(String[] args) throws ParseException, RuntimeException {
     }
 
     public Method getMethod(){
+    	Preconditions.checkNotNull(method, "Method cannot be null");
```

Comment:
These checks are not appropriate in a getter. Do these checks during construction if nothing is allowed to be null.

Reply:
Those fields are not set during the constructor but actually when `parse()` is called.
This check was actually to be sure that the getter is called after `parse()` so I change the precondition to `check state()` to reflect this.

Reply:
There are two generals rules we should try to satisfy at all times:
- An object always has valid state (simple accesses to query its state via getters should not cause problems). If getters throw exceptions, a caller cannot even inspect the objects state without causing problems. That is a big no-no.
- Validation should happen as early as possibly (ideally during construction, if possible)

In this case, the `XOptions` classes are always used by first calling the constructor and then immediately calling its parse method. Before calling `parse`, the object is basically in an invalid state, therefore`parse` should be part of the construction. Simply make the constructor and the `parse` method protected and then offer a static factory method `fromArgs` to construct a fully valid object. Then you do not need to do any checks inside the getters and the callers code simplifies.